pkg/compactor/block_upload_test.go

```diff
@@ -1591,3 +1813,64 @@ func TestMultitenantCompactor_GetBlockUploadStateHandler(t *testing.T) {
 		})
 	}
 }
+
+// marshalAndUploadJSON is a test helper for uploading a meta file to a certain path in a bucket.
+func marshalAndUploadJSON(t *testing.T, bkt objstore.Bucket, pth string, val interface{}) {
+	t.Helper()
+	err := marshalAndUploadToBucket(context.Background(), bkt, pth, val)
+	require.NoError(t, err)
+}
+
+// populateMetaFileList populates the thanos files list in the meta file based on files in a block directory.
+func populateMetaFileList(t *testing.T, dir string, meta *metadata.Meta) {
```

Comment:
Can we use `block.GatherFileStats(dir)` function?

Reply:
yes, I need to start searching existing code before I try to re-invent the wheel